utils.js

```diff
@@ -165,3 +165,38 @@ class ObjectsMap {
         });
     }
 };
+
+let _currentDesktopsMatches = {};
+// is:
+// @name: desktop string you want to assert if it matches the current desktop env
+//
+// The function examples XDG_CURRENT_DESKTOP and return if the current desktop
+// is part of that desktop string.
+//
+// Return value: if the environment isn't set or doesn't match, return False
+// otherwise, return True.
+//
+// This function is a copy of:
+// https://github.com/endlessm/gnome-shell/blob/master/js/misc/desktop.js
+function is(name) {
+    if (_currentDesktopsMatches[name] !== undefined) {
+        return _currentDesktopsMatches[name];
+    }
+
+    let desktopsEnv = GLib.getenv('XDG_CURRENT_DESKTOP');
+    if (!desktopsEnv) {
+        _currentDesktopsMatches[name] = false;
+        return false;
+    }
+
+    let desktops = desktopsEnv.split(":");
```

Comment:
This can be simplified as:

```javascript
const hasMatch = desktopEnv.split(':').some(desktop => desktop === name);
_currentDesktopsMatches[name] = hasMatch;
return hasMatch;
```

Reply:
Done, I've replaced the code with this one.